include/xtensor/xbuilder.hpp

```diff
@@ -719,42 +762,56 @@ namespace xt
      * \endcode
      */
     template <class E>
-    inline auto diag(E&& arr)
+    inline auto diag(E&& arr, int k = 0)
     {
         using CT = xclosure_t<E>;
-        std::size_t s = arr.shape()[0];
-        return detail::make_xgenerator(detail::fn_impl<detail::diag_fn<CT>>(detail::diag_fn<CT>(std::forward<E>(arr))),
+        std::size_t s = arr.shape()[0] + std::abs(k);
+        return detail::make_xgenerator(detail::fn_impl<detail::diag_fn<CT>>(detail::diag_fn<CT>(std::forward<E>(arr), k)),
                                        { s, s });
     }
 
     /**
-     * @brief Flip xexpression in the left/right direction. Essentially flips the last axis.
+     * @brief Reverse the order of elements in an xexpression along the given axis.
+     * 
+     * @param arr the input xexpression
+     * @param axis the axis along which elements should be reversed
      *
-     * @param arr the input array
-     * @returns xexpression with values flipped in left/right direction
+     * @return xexpression evaluating to reversed array
      */
     template <class E>
-    inline auto fliplr(E&& arr)
+    inline auto flip(E&& arr, std::size_t axis)
     {
         using CT = xclosure_t<E>;
         auto shape = arr.shape();
-        return detail::make_xgenerator(detail::fliplr_impl<CT>(std::forward<E>(arr)),
+        return detail::make_xgenerator(detail::flip_impl<CT>(std::forward<E>(arr), axis),
                                        shape);
     }
 
+
     /**
-     * @brief Flip xexpression in the up/down direction. Essentially flips the last axis.
+     * @brief Flip xexpression in the left/right direction.
+     * This is equivalent to xt::flip(arr, 1);
+     *
+     * @param arr the input array
+     * @returns xexpression with values flipped in left/right direction
+     */
+    template <class E>
+    inline auto fliplr(E&& arr)
+    {
+        return flip(std::forward<E>(arr), 1);
+    }
```

Comment:
sure, let's remove it. we can also add `fliplr -> flip(a, 1)` to the cheatsheet i guess.